### ISSUE TITLE
Discussion: Generic Guns, 'Archetypal guns'

### ISSUE THREAD

**Issue by tenmillimaster** [CleverRaven/Cataclysm-DDA#35483]
**The Issue**  
Generic guns is an accessibility mod that is in need of updating. In the process of doing so, we need to identify which 'archetypes' a firearm can be distilled to, without losing nuance while also retaining simplicity.

Rather than leave some item id's untouched, as the mod currently does, we'd like to migrate all guns to one of these archetypal guns. 

**Archetypal Calibers**  
These are the calibers I would distill all 'firearms' down to in game:
![image](https://user-images.githubusercontent.com/7764202/68673833-371a9900-051a-11ea-9a82-9a302697da41.png) 
Each column contains core game calibers that would be migrated to their column header's caliber,  e.g. all 10mm, .45 acp, and 9mm would migrate to 'pistol ammo'.

**Archetypal firearms**
Loosely arranged by caliber and 'format', this is a matrix of firearms that would be included in game. The format doesn't really make sense for the launchers, however.
![image](https://user-images.githubusercontent.com/7764202/68674021-a98b7900-051a-11ea-8d4b-ba6de6740dd6.png)

**Paring Down**  
A number of these firearms are quite close, e.g. the battle rifle and sniper rifle in 'rifle', the sporter rifle and assault rifle in 'light rifle', flintlocks and percussion weapons...
I do not use this mod myself, so would like feedback on if there are sufficient firearms, or too many. And if too many, which do you feel should be removed?

**Comment by mlangsdorf** [comment]
target pistol (tiny rifle magazine fed handgun)
small pistol (small pistol magazine fed handgun)
large pistol (large pistol magazine fed handgun)
small revolver (small pistol revolver handgun)
hunting revolver (large pistol or black powder revolver handgun)
antique revolver (black powder revolver handgun)
antique pistol (black powder single shot handgun)
smg (large pistol magazine fed submachine gun - nobody using GG cares about the distinctions between SMGs and PDWs)
plinking rifle (tiny rifle longarm with a fixed magazine)
assault rifle (large rifle longarm with detachable magazine and automatic fire)
hunting rifle (large rifle long arm with detachable magazine)
cowboy rifle (large pistol longarm with fixed magazine)
survivalist carbine (large pistol longarm with detachable magazine)
sniper rifle (large rifle long arm with fixed magazine)
anti-material rifle (huge rifle longarm with detachable magazine)
elephant gun (huge rifle longarm with 2 shots)
antique musket (black powder, single shot longarm)
double-barrel shotgun (blackpowder or shotgun long with two shots)
shotgun (blackpowder or shotgun longarm with fixed magazine)
light machine gun (large rifle longarm that takes ammo belts)
heavy machine gun (huge rifle longarm that takes ammo belts)
pipe pistol (large pistol single shot handgun)
pipe rifle (large rifle single shot longarm)
flare pistol (grenade single shot handgun)
grenade launcher (grenade single shot longarm)
revolver grenade launcher (grenade longarm with 12 shot fixed magazine)
automatic grenade launcher (grenade longarm with belt feed)
disposable rocket launcher (rocket single use longarm)
"bazooka" (rocket reloadable single use longarm)

ammo types are:
small pistol (.32 ACP): anything smaller than 9mm
large pistol (.45 ACP): 9mm or heavier
tiny rifle (.22LR): .22LR and variants
large rifle (.308): any rifle round below .375 H&H
huge rifle (.50 BMG): .375 H&H and larger rifle rounds
shotgun (12g shot): any shotshell
black powder (paper cartridge): any black powder load
grenade (40mm grenade): any flare or grenade round
rocket (84mm carl gustav rocket): any rocket

This is slightly fewer categories than you had; I don't think the distinction between cap and ball, flintlock, and blunderbuss matters, and all rifle rounds between .22LR and .50 BMG are just rifle rounds.

**Comment by Night-Pryanik** [comment]
> Generic guns is an accessibility mod

We use "accessibility" term only when describing how to help people with physical problems (blind, color-blind etc) to play the game. "Generic guns" is a mod purely for personal tastes, it has nothing to do with accessibility.

**Comment by tenmillimaster** [comment]
Oh, my bad. A more senior developer than me had referred to it in such a manner, but I suppose that was just an opinion.

**Comment by I-am-Erk** [comment]
We're really not that anal retentive about the tags, and I'd absolutely call this an accessibility change as it makes the game much more accessible to a wide range of players

**Comment by I-am-Erk** [comment]
Regarding the actual question, I note you're missing single shot rifles, like the classic bolt action hunting rifle?

**Comment by kevingranade** [comment]
I've gotten serious feedback from a number of people that they find guns prohibitively difficult to interact with without this mod.  IMO that qualifies.

**Comment by I-am-Erk** [comment]
That list includes me. I've managed to learn a couple types, but by and large I just avoid them.

**Comment by tenmillimaster** [comment]
@I-am-Erk in both mine and Mlangsdorf's previews, a  bolt action hunting rifle would be a sniper rifle.

**Comment by ymber** [comment]
I'd keep the items and ammotypes pretty minimal. I quite like the direction of Tonkatsu's list but I'd cut it down so there's no role overlap between the weapons.

- Remove
    - cap and ball ammunition and weapons
    - double carbine
    - double barrel shotgun
    - backup pistol
    - plinking carbine
    - lever action carbine
    - elephant gun
- Replace
    - M1014 with Saiga-12
    - M110A1 with M24
- Add
    - machine pistol (Glock 18C)
    - hunting rifle without optics (mosin 91/30)

**Comment by I-am-Erk** [comment]
I don't think that's all that necessary... Generic guns doesn't mean removing the gameplay of weapon selection entirely, the goal is mainly to remove the part where you have to memorize a bunch of letter-number string brand names that offer no information about the weapon. Having some choice between a few weapons is still desirable I think.

**Comment by ymber** [comment]
Closing this as a concluded discussion issue.
